src/pages/index.js

```diff
@@ -84,24 +90,148 @@ function handleImageClick(data) {
 function handleProfileEditSubmit(formValues) {
   userInfo.setUserInfo({
     name: formValues.title,
-    about: formValues.about,
+    about: formValues.description,
   });
+  editProfileModal.setLoading(true);
+  api
+    .setUserInfo(formValues.title, formValues.description)
+    .then((res) => {
+      userInfo.getUserInfo(res.name, res.about);
+      editProfileModal.setLoading(false);
+    })
+    .catch((err) => {
+      console.error("Error updating user info", err);
+      alert(err);
+    });
   editProfileModal.close();
 }
 function handleAddCardFormSubmit(formValues) {
   const name = formValues.title;
   const link = formValues.link;
+  // Make API request to upload card
+  addCardModal.setLoading(true);
+  api
+    .uploadCard({ name, link })
+    .then((cardData) => {
+      const card = createCard(cardData);
+      addCardModal.setLoading(false);
 
-  const card = createCard({ name, link });
-  cardList.addItem(card);
-  console.log(formValues);
-  cardAddForm.reset();
-  addCardModal.close();
+      cardList.addItem(card);
+      addCardModal.close();
+      cardAddForm.reset();
+    })
+    .catch((error) => {
+      console.error(error);
+    });
 }
 
 function createCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick);
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCardSubmit,
+    likeCard,
+    unlikeCard
+  );
   return card.generateCard();
 }
 
-//Event Listeners
+api
+  .getInitialCards()
+  .then((res) => {
+    console.log(res);
+    cardList.renderItems(res);
+  })
+
+  .catch((err) => alert(err));
+
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo(res);
+    userInfo.updateProfileImage(res);
+  })
+  .catch((err) => alert(err));
+
+//Avatar
+const profileImageForm = document.querySelector("#edit-avatar-form");
+const profileFormValidator = new FormValidator(
```

Comment:
**COULD BE IMPROVED**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```